src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -221,47 +229,81 @@ static SGAAssemblyResult runSGAModulesInSerial(final String sgaPath,
         }
 
-        final File fileToAssemble   = new File(tempWorkingDir, fileNameToAssemble);
-        final String contigsFileName = runAndStopEarly("assemble", fileToAssemble, sga, tempWorkingDir, indexer, indexerArgs, runtimeInfo, enableSTDIOCapture);
-        if(null == contigsFileName){
-            return new SGAAssemblyResult(null, runtimeInfo);
+        {//overlap
+            moduleArgs.clear();
+            moduleArgs.add("--min-overlap");   moduleArgs.add(String.valueOf(MIN_OVERLAP_IN_FILTER_OVERLAP_ASSEMBLE));
+            moduleArgs.add("--error-rate");    moduleArgs.add(String.valueOf(OVERLAP_STEP_ERROR_RATE));
+
+            nameOfFileToFeedToNextStep = runSGAModule(sgaPath, "overlap", ".asqg.gz", new File(workingDir, nameOfFileToFeedToNextStep), workingDir, null, null, moduleArgs, runtimeInfo, enableSTDIOCapture);
+            if( null == nameOfFileToFeedToNextStep ){ return new SGAAssemblyResult(null, runtimeInfo); }
+        }
+
+        {//assembly
+            moduleArgs.clear();
+            moduleArgs.add("--cut-terminal");         moduleArgs.add(String.valueOf(CUT_OFF_TERMINAL_BRANCHES_IN_N_ROUNDS));
+            moduleArgs.add("--max-gap-divergence");   moduleArgs.add(String.valueOf(ASSEMBLE_STEP_MAX_GAP_DIVERGENCE));
+            moduleArgs.add("--min-overlap");          moduleArgs.add(String.valueOf(MIN_OVERLAP_IN_FILTER_OVERLAP_ASSEMBLE));
+            moduleArgs.add("--min-branch-length");    moduleArgs.add(String.valueOf(ASSEMBLE_STEP_MIN_BRANCH_TAIL_LENGTH));
+            moduleArgs.add("--out-prefix");           moduleArgs.add(nameOfFileToFeedToNextStep.replace(".asqg.gz", ""));
+            if(TURN_ON_TRANSITIVE_REDUCTION){
+                moduleArgs.add("--transitive-reduction");
+            }
+
+            nameOfFileToFeedToNextStep = runSGAModule(sgaPath, "assemble", "-contigs.fa", new File(workingDir, nameOfFileToFeedToNextStep), workingDir, null, null, moduleArgs, runtimeInfo, enableSTDIOCapture);
+            if( null == nameOfFileToFeedToNextStep ){
+                return new SGAAssemblyResult(null, runtimeInfo);
+            }else{
+                nameOfFileToFeedToNextStep = nameOfFileToFeedToNextStep.replace(".asqg", ""); // strip out substring
+            }
         }
 
         logger.debug("SGAOnSpark_debug: finished actual assembly process for " + rawFASTQFile.getName() + " at " + dateFormat.format(new Date()));
 
         // if code reaches here, all steps in the SGA pipeline went smoothly,
         // but the following conversion from File to ContigsCollection may still err
-        final File assembledContigsFile = new File(tempWorkingDir, contigsFileName);
+        final File assembledContigsFile = new File(workingDir, nameOfFileToFeedToNextStep);
         final List<String> contigsFASTAContents = (null==assembledContigsFile) ? null : Files.readAllLines(Paths.get(assembledContigsFile.getAbsolutePath()));
```

Comment:
IDEA brings up the fact that `assembledContigsFile` cannot be null here since you just assigned it to a `new File()`. I think you might want to check `assembledContigsFile.exists()`?